worker/deployer/nested_test.go

```diff
@@ -239,19 +264,68 @@ func (s *NestedContextSuite) TestErrTerminateAgentFromAgentWorker(c *gc.C) {
 			c.Fatal("unit not stopped")
 		}
 	}
+	return report
+}
 
-	c.Assert(ctx.Report(), jc.DeepEquals, map[string]interface{}{
-		"deployed": []string{"something/0"},
-		"stopped":  []string{"something/0"},
-		"units": map[string]interface{}{
-			"workers": map[string]interface{}{},
-		},
+func (s *NestedContextSuite) TestStopStartUnits(c *gc.C) {
+	ctx := s.newContext(c)
+	s.deployThreeUnits(c, ctx)
+
+	handled := s.hub.Publish(message.StopUnitTopic, message.Units{
+		Names: []string{"first/0", "second/0", "unknown/2"},
 	})
+	s.waitForEventHandled(c, handled)
+	report := ctx.Report()
+	c.Assert(report["stopped"], jc.DeepEquals, []string{"first/0", "second/0"})
+
+	// Start one back up again.
+	handled = s.hub.Publish(message.StartUnitTopic, message.Units{
+		Names: []string{"first/0", "third/0", "unknown/2"},
+	})
+	s.waitForEventHandled(c, handled)
+	report = ctx.Report()
+	c.Assert(report["stopped"], jc.DeepEquals, []string{"second/0"})
 }
 
-func (s *NestedContextSuite) TestReport(c *gc.C) {
+func (s *NestedContextSuite) TestUnitStatus(c *gc.C) {
+	responseHandled := make(chan struct{})
+	unsub := s.hub.Subscribe(message.UnitStatusResponseTopic, func(_ string, payload interface{}) {
+		response := payload.(message.Status) // TODO rename to unit status
+		c.Check(response, jc.DeepEquals, message.Status{
+			"agent": "machine-42",
+			"units": map[string]string{
+				"first/0":  "running",
+				"second/0": "stopped",
+				"third/0":  "running",
+			},
+		})
+		close(responseHandled)
+	})
+	defer unsub()
+
 	ctx := s.newContext(c)
+	s.deployThreeUnits(c, ctx)
+	// And stop one.
+	handled := s.hub.Publish(message.StopUnitTopic, message.Units{
```

Comment:
Would like to see some tests for when Start/StopUnitTopic etc is called twice

Reply:
Sure...